R/html_document.R

```diff
@@ -434,6 +449,20 @@ html_document <- function(toc = FALSE,
     args
   }
 
+  # post-processor that use the output file from pandoc
+  post_processor <- function(metadata, input_file, output_file, clean, verbose) {
```

Comment:
This won't work with output formats that extend `html_document()` but just replace the post processor (instead of wrapping it). I don't know if that's common but I think it's worth noting and thinking about.

Reply:
you mean a format doing something like 

```r
new_format <- function(...) {
    base_format <- html_document(...)
    base_format$post_processor <- new_postproc() 
    base_format
}
```` 
instead of `output_format` and its `base_format` argument ?

Reply:
Right.

Reply:
If someone overwrite the post processor AND use `highlight_downlit = TRUE`, there won't be any downlit processing but the highlight would still default to "a11y" and not highlightjs. But not sure we can detect this specific case.
I am not sure to see what we could do here. If someone extend `html_document` but replace the post processor. Currently, `html_document_base` post processor will also be replaced - and some feature like `copy_resource` are lost. 
And if some extend `html_document` AND use `highlight_downlit = TRUE`, we could think the post processor would have been kept. 

I believe, if someone extend on the feature of `html_document`, it would be the responsibility of the author to keep the post processor if needed. Different formats not using `output_formats` mechanism are doing it already with : 
```r
new_format <- function(...) {
    base_format <- html_document(...)
    post <- base_format$post_processor
    base_format$post_processor <- function(..., output_file) {
        if (is.function(post)) output_file <- post(...)
        new_proc(..., output_file)
    }
    base_format
}
```

I believe a vignette (or [recipe](https://github.com/yihui/rmarkdown-cookbook/issues/285)) on how to extend rmarkdown format would be necessary to share the good practices and the pitfall to watch out for.

Reply:
I agree with your points; I think it's important just to note that this has the potential to break downstream formats. I don't think this is particularly important here, but I think it's the sort of analysis that's necessary when making changes to old and popular functions.

Reply:
You're right, we need to check that closely. But I am worried we do not have any counter measure if you extend based on a format but replace its internal processing. That is why the `output_formats` function exists and is exported in the first place I believe, because it contains a merging strategy with the base format provided (so you can't overwrite by mistake)